src/dask_histogram/core.py

```diff
@@ -1030,17 +1026,11 @@ def __call__(self, a: AggHistogram, b: AggHistogram) -> AggHistogram:
         deps = []
         if is_dask_collection(a):
             deps.append(a)
-            k1 = a.name
-        else:
-            k1 = a  # type: ignore
         if is_dask_collection(b):
             deps.append(b)
-            k2 = b.name
-        else:
-            k2 = b  # type: ignore
-        k1 = a.__dask_tokenize__() if is_dask_collection(a) else a  # type: ignore
-        k2 = b.__dask_tokenize__() if is_dask_collection(b) else b  # type: ignore
-        llg = {name: (self.func, k1, k2)}
+        k1 = a.__dask_keys__()[0] if is_dask_collection(a) else a  # type: ignore
+        k2 = b.__dask_keys__()[0] if is_dask_collection(b) else b  # type: ignore
+        llg = {(name, 0): (self.func, k1, k2)}
```

Comment:
This could be expressed as:
```python3
        llg = blockwise(
            self.func,
            name,
            "i",
            a.name, "i", b.name, "i",
            numblocks={a.name: (1, ), b.name: (1, )},
        )
```
and then we remove all `MaterializedLayer` instances that dask_histogram produces, but it seems we found where the real problem lies.

This would probably be good for robustness though?

Reply:
@martindurant over / under on using the blockwise on a known single partition layer vs. a MaterializedLayer?

Reply:
I would vote against blockwise for this one, as it rather obscures what we are doing, and simple is good

Reply:
Indeed, it doesn't seem to affect optimization time or results so no need.